src/posix/platform/multicast_routing.cpp

```diff
@@ -432,9 +438,9 @@ const char *MulticastRoutingManager::MifIndexToString(MifIndex aMif)
     return string;
 }
 
+#if OPENTHREAD_CONFIG_LOG_PLATFORM && OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_DEBG
 void MulticastRoutingManager::DumpMulticastForwardingCache(void) const
 {
-#if OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_DEBG
```

Comment:
Why not add `OPENTHREAD_CONFIG_LOG_PLATFORM` here?

Reply:
I add `OPENTHREAD_CONFIG_LOG_PLATFORM && OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_DEBG` to wrap the whole methods. Thoughts?

Reply:
Seems we are adding more code than necessary. Wrapping the body of the function seems already enough?

Reply:
I also agree with @simonlingoogle 's suggestion. Better to avoid defining new macros.

Reply:
SGTM. I've updated and only adds `OPENTHREAD_CONFIG_LOG_PLATFORM`. @abtink @jwhui @simonlingoogle